### ISSUE TITLE
Go 1.8 step-by-step execution panics

### ISSUE THREAD

**Issue by dlsniper** [derekparker/delve#698]
1. What version of Delve are you using (`dlv version`)?
63b8fa82698bc9bcf8380c0304fe290047d68240
2. What version of Go are you using? (`go version`)?
1.8 beta 2
3. What operating system and processor architecture are you using?
Windows 10 Pro, build 14393
4. What did you do?
```go
package demo_test

import (
	"fmt"
	"testing"
)

func TestDebug(t *testing.T) {
	a := "value" // set breakpoint
	fmt.Printf("%s\n", a)
}
```

Set a breakpoint on the assignment, then use step by step execution to run the code, when the function will exit the application will terminate with the following panic:
```
unexpected fault address 0xc042d6bf90
fatal error: fault
[signal 0xc0000005 code=0x1 addr=0xc042d6bf90 pc=0x4fdd61]

goroutine 19 [running]:
runtime.throw(0x542f69, 0x5)
	C:/Go1.8/src/runtime/panic.go:596 +0x9c fp=0xc0420abec8 sp=0xc0420abea8
runtime.sigpanic()
	C:/Go1.8/src/runtime/signal_windows.go:164 +0xff fp=0xc0420abef8 sp=0xc0420abec8
github.com/dlsniper/u/gog/3227_test.TestDebug(0xc04204c4e0)
	D:/go/src/github.com/dlsniper/u/gog/3227/3227_test.go:10 +0xd1 fp=0xc0420abfa8 sp=0xc0420abef8
testing.tRunner(0xc04204c4e0, 0x54ce88)
	C:/Go1.8/src/testing/testing.go:656 +0x9a fp=0xc0420abfd0 sp=0xc0420abfa8
runtime.goexit()
	C:/Go1.8/src/runtime/asm_amd64.s:2184 +0x1 fp=0xc0420abfd8 sp=0xc0420abfd0
created by testing.(*T).Run
	C:/Go1.8/src/testing/testing.go:693 +0x2cb

goroutine 1 [chan receive]:
testing.(*T).Run(0xc04204c410, 0x543962, 0x9, 0x54ce88, 0xc042033d28)
	C:/Go1.8/src/testing/testing.go:694 +0x2f5
testing.runTests.func1(0xc04204c410)
	C:/Go1.8/src/testing/testing.go:877 +0x6e
testing.tRunner(0xc04204c410, 0xc042033de8)
	C:/Go1.8/src/testing/testing.go:656 +0x9a
testing.runTests(0xc04203cd80, 0x5eeaa0, 0x1, 0x1, 0x0)
	C:/Go1.8/src/testing/testing.go:883 +0x2a4
testing.(*M).Run(0xc0420407e0, 0x60f368)
	C:/Go1.8/src/testing/testing.go:818 +0x103
main.main()
	github.com/dlsniper/u/gog/3227/_test/_testmain.go:40 +0xde
```

Not really sure if this is related to not having https://github.com/derekparker/delve/pull/667 in delve or not yet.
Hope it helps.

Also this seems to not happen when using Go 1.7.4 on the same system but I'm not sure if this is relevant or not.

**Comment by dlsniper** [comment]
Originally from here: https://youtrack.jetbrains.com/issue/GO-3227

**Comment by doctor-fate** [comment]
Same on Ubuntu 16.04

**Comment by aarzilli** [comment]
I'll check but this sounds very much like something I fixed in #667.

**Comment by aarzilli** [comment]
Duplicate of #664.
